connector/src/test/scala/com/microsoft/kusto/spark/KustoSinkBatchE2E.scala

```diff
@@ -336,37 +336,79 @@ class KustoSinkBatchE2E extends AnyFlatSpec with BeforeAndAfterAll {
-    val engineKcsb = ConnectionStringBuilder.createWithAadAccessTokenAuthentication(
-      kustoTestConnectionOptions.cluster,
-      kustoTestConnectionOptions.accessToken)
-    val kustoAdminClient = ClientFactory.createClient(engineKcsb)
-    kustoAdminClient.execute(
-      kustoTestConnectionOptions.database,
-      generateTempTableCreateCommand(table, columnsTypesAndNames = "ColA:string, ColB:int"))
-
-    df.write
-      .format("com.microsoft.kusto.spark.datasource")
-      .partitionBy("value")
-      .option(KustoSinkOptions.KUSTO_CLUSTER, kustoTestConnectionOptions.cluster)
-      .option(KustoSinkOptions.KUSTO_DATABASE, kustoTestConnectionOptions.database)
-      .option(KustoSinkOptions.KUSTO_TABLE, table)
-      .option(KustoSinkOptions.KUSTO_ACCESS_TOKEN, kustoTestConnectionOptions.accessToken)
-      .option(KustoSinkOptions.KUSTO_TIMEOUT_LIMIT, (8 * 60).toString)
-      .mode(SaveMode.Append)
-      .save()
+  private val ingestTests =
+    Table(
+      "testName",
+      "DMStorage",
+      "CustomStorage"
+    )
+
+
+  TableDrivenPropertyChecks.forAll (ingestTests) { testName =>
+    "KustoBatchSinkSync" should s"also ingest simple data to a Kusto cluster for $testName" taggedAs KustoE2E in {
+      import spark.implicits._
+      var skipAssertions = false
+      val df = rows.toDF("name", "value")
+      val prefix = s"KustoBatchSinkE2E_Ingest_${testName}"
+      val table = KustoQueryUtils.simplifyName(s"${prefix}_${UUID.randomUUID()}")
+      val engineKcsb = ConnectionStringBuilder.createWithAadAccessTokenAuthentication(
+        kustoTestConnectionOptions.cluster,
+        kustoTestConnectionOptions.accessToken)
+      val kustoAdminClient = ClientFactory.createClient(engineKcsb)
+      kustoAdminClient.execute(
+        kustoTestConnectionOptions.database,
+        generateTempTableCreateCommand(table, columnsTypesAndNames = "ColA:string, ColB:int"))
+
+      if (testName == "DMStorage") {
+        df.write
+          .format("com.microsoft.kusto.spark.datasource")
+          .partitionBy("value")
+          .option(KustoSinkOptions.KUSTO_CLUSTER, kustoTestConnectionOptions.cluster)
+          .option(KustoSinkOptions.KUSTO_DATABASE, kustoTestConnectionOptions.database)
+          .option(KustoSinkOptions.KUSTO_TABLE, table)
+          .option(KustoSinkOptions.KUSTO_ACCESS_TOKEN, kustoTestConnectionOptions.accessToken)
+          .option(KustoSinkOptions.KUSTO_TIMEOUT_LIMIT, (8 * 60).toString)
+          .mode(SaveMode.Append)
+          .save()
+      } else {
+        // Use custom storage
+        val storageUrl = System.getProperty("INGEST_STORAGE_URL")
+        val containerName = System.getProperty("INGEST_STORAGE_CONTAINER")
+        if(StringUtils.isEmpty(storageUrl) || StringUtils.isEmpty(containerName)) {
+          skipAssertions = true
+          KDSU.logWarn(
+            className,
+            "No ingestion storage URL or container name provided. Skipping ingestion test.")
```

Comment:
please either comment out the test if its not ready or run it always - lets not ignore on wrong setup